pkg/acquisition/modules/kubernetesaudit/k8s_audit.go

```diff
@@ -0,0 +1,200 @@
+		return errors.Wrap(err, "Cannot parse k8s-audit configuration")
+	}
+
+	ka.config = k8sConfig
+
+	if ka.config.ListenAddr == "" {
+		return fmt.Errorf("listen_addr cannot be empty")
+	}
+
+	if ka.config.ListenPort == 0 {
+		return fmt.Errorf("listen_port cannot be empty")
+	}
+
+	if ka.config.WebhookPath == "" {
+		return fmt.Errorf("webhook_path cannot be empty")
+	}
+
+	if ka.config.WebhookPath[0] != '/' {
+		ka.config.WebhookPath = "/" + ka.config.WebhookPath
+	}
+
+	if ka.config.Mode == "" {
+		ka.config.Mode = configuration.TAIL_MODE
+	}
+	return nil
+}
+
+func (ka *KubernetesAuditSource) Configure(config []byte, logger *log.Entry) error {
+	ka.logger = logger
+
+	ka.logger.Tracef("K8SAudit configuration: %+v", ka.config)
+
+	ka.addr = fmt.Sprintf("%s:%d", ka.config.ListenAddr, ka.config.ListenPort)
+
+	ka.mux = http.NewServeMux()
+
+	ka.server = &http.Server{
+		Addr:    ka.addr,
+		Handler: ka.mux,
+	}
+
+	ka.mux.HandleFunc(ka.config.WebhookPath, ka.webhookHandler)
+	return nil
+}
+
+func (ka *KubernetesAuditSource) ConfigureByDSN(dsn string, labels map[string]string, logger *log.Entry) error {
+	return fmt.Errorf("k8s-audit datasource does not support command-line acquisition")
+}
+
+func (ka *KubernetesAuditSource) GetMode() string {
+	return ka.config.Mode
+}
+
+func (ka *KubernetesAuditSource) GetName() string {
+	return "k8s-audit"
+}
+
+func (ka *KubernetesAuditSource) OneShotAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	return fmt.Errorf("k8s-audit datasource does not support one-shot acquisition")
+}
+
+func (ka *KubernetesAuditSource) StreamingAcquisition(out chan types.Event, t *tomb.Tomb) error {
+	ka.outChan = out
+	t.Go(func() error {
+		defer types.CatchPanic("crowdsec/acquis/k8s-audit/live")
+		ka.logger.Infof("Starting k8s-audit server on %s:%d%s", ka.config.ListenAddr, ka.config.ListenPort, ka.config.WebhookPath)
+		t.Go(func() error {
+			err := ka.server.ListenAndServe()
+			if err != nil && err != http.ErrServerClosed {
+				return errors.Wrap(err, "k8s-audit server failed")
+			}
+			return nil
+		})
+		<-t.Dying()
+		ka.logger.Infof("Stopping k8s-audit server on %s:%d%s", ka.config.ListenAddr, ka.config.ListenPort, ka.config.WebhookPath)
+		ka.server.Shutdown(context.TODO())
+		return nil
+	})
+	return nil
+}
+
+func (ka *KubernetesAuditSource) CanRun() error {
+	return nil
+}
+
+func (ka *KubernetesAuditSource) Dump() interface{} {
+	return ka
+}
+
+func (ka *KubernetesAuditSource) webhookHandler(w http.ResponseWriter, r *http.Request) {
+	requestCount.WithLabelValues(ka.addr).Inc()
+	if r.Method != http.MethodPost {
+		w.WriteHeader(http.StatusMethodNotAllowed)
+		return
+	}
+	ka.logger.Tracef("webhookHandler called")
+	var auditEvents audit.EventList
+
+	jsonBody, err := io.ReadAll(r.Body)
+	if err != nil {
+		ka.logger.Errorf("Error reading request body: %v", err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	ka.logger.Tracef("webhookHandler receveid: %s", string(jsonBody))
```

Comment:
shouldn't we have a `.Debugf` version ? Might be useful to debug the datasource ?